integration_tests/steps/two_sided_market.py

```diff
@@ -185,3 +185,17 @@ def cep_offer_residual_offer_rate(context):
             for id, offer in market.offers.items():
                 assert isclose((offer.price / offer.energy),
                                cep.strategy.energy_rate[market.time_slot])
+
+
+@then('Energy producer is {producer} & consumer is {consumer}')
+def energy_origin(context, producer, consumer):
+    trade_count = 0
+
+    for market in context.simulation.area.past_markets:
```

Comment:
Are you sure that you are looping over all past markets? Is KEEP_PAST_MARKETS set to True?

Reply:
@hannesdiedrich it is set to true by default for the integration tests, in order to simplify checks. Have a look at environment.py -> before_scenario and before_all methods.

Reply:
Ah, Thanks!